src/main/java/de/tum/in/www1/artemis/domain/MathSubmission.java

```diff
@@ -0,0 +1,93 @@
+package de.tum.in.www1.artemis.domain;
+
+import static de.tum.in.www1.artemis.config.Constants.MAX_SUBMISSION_TEXT_LENGTH;
+
+import javax.persistence.*;
+import javax.validation.constraints.Size;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.enumeration.Language;
+
+/**
+ * A MathSubmission.
+ */
+@Entity
+@DiscriminatorValue(value = "MATH")
+@SecondaryTable(name = "math_submission_details")
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class MathSubmission extends Submission {
+
+    @Override
+    public String getSubmissionExerciseType() {
+        return "math";
+    }
+
+    private static final int MAX_EXCERPT_LENGTH = 100;
+
+    @Column(name = "text", table = "math_submission_details")
+    @Size(max = MAX_SUBMISSION_TEXT_LENGTH, message = "The text submission is too large.")
+    private String text;
```

Comment:
This column already exists as part of the `submission` base table, so you should reuse that one, instead of creating your own.

Reply:
That's actually a issue w/ the `TextSubmission` base table, as it does define a `@SecondaryTable`, with the evident intention to store text-exercise specific fields (`text`, `language`). It happens to be, that specifying the `table` in the column definitions on `TextSubmission` was forgotten, leaving us with an empty `text_submission_details` table.

But yea, I guess we have to deal with that, and I'll just remove the `@SecondaryTable` from math exercises, live with it that it exists for text exercises, and just re-use the `text` field on the submissions table.